core/field.js

```diff
@@ -441,15 +446,24 @@ Blockly.Field.prototype.updateColour = function() {
 };
 
 /**
- * Draws the border with the correct width.
- * Saves the computed width in a property.
+ * Used by getSize() to move/resize any dom elements, and get the new size.
+ *
+ * All rendering that has an effect on the size/shape of the block should be
+ * done here, and should be triggered by getSize().
  * @protected
  */
 Blockly.Field.prototype.render_ = function() {
   // Replace the text.
-  this.textElement_.textContent = this.getDisplayText_();
+  try {
+    this.textElement_.childNodes[0].nodeValue = this.getDisplayText_();
+  } catch (error) {
+    if (this.textElement_.childNodes.length == 0) {
```

Comment:
Why do this as a try-catch instead of checking the length of `childNodes` before trying to make the change?

Reply:
So this happened because of how the angle field's degree symbol is static rather than reappended every time render is called. It used to be reappended everytime because when the textElement_'s textContent was set it would get kicked. So instead of setting the textContent it just sets the nodeValue of the textElement_'s first text node. Buuut if the text element doesn't have a first text node everything breaks. This can happen if a developer writes their initView function incorrectly.

TL:DR A series of changes meant I had to add this so developers know if they derp something up.

Really this needs a better solution though, how would you feel about some sort of textGroup_ that the textElement_ as well as things like degree symbols could be children of? Then render could go back to how it was before, and this could be removed.

Reply:
Opinions @rachel-fenichel ?

Reply:
Either that, or somehow move the special sauce back into field angle (even if it means putting it back to a dynamic state).  I'd like to avoid having a throw like this in the general field render path.

Reply:
I ended up reverting it because I changing it to have a new DOM structure would mean reworking some CSS, and I didn't want to do that in this PR.

I created an [issue ](https://github.com/google/blockly/issues/2484)to track fixing this (because it bugs me), but if you don't think it's a big deal I'm fine with closing it.